### DISCUSSION TITLE
Why does FileUpload try to upload directly to S3?

### DISCUSSION THREAD

**Discussion by NicolasSacC** [filamentphp/filament#7847] [Help] [answered]
Hi!

Still on Filament 2

I got a weird issue: Filament tries to upload to S3 directly from the browser instead of doing it server side. Obviously, it doesn't work.

Why does it happen like that?

Here is my code: 
```
                Forms\Components\FileUpload::make('image')
                    ->disk('s3')
                    ->visibility('public')
```

.env: 
```
AWS_ACCESS_KEY_ID=MYAWSID
AWS_SECRET_ACCESS_KEY=MYAWSKEY
AWS_DEFAULT_REGION=ca-central-1
AWS_BUCKET=my-bucket
AWS_USE_PATH_STYLE_ENDPOINT=false
FILESYSTEM_DISK=s3
```

And here is a screenshot:

![Screenshot 2023-08-17 at 08 03 00](https://github.com/filamentphp/filament/assets/16766008/5b966159-392c-47ca-9bd5-b2482d602a26)

**Comment by danharrin** [comment, marked as answer]
> I got a weird issue: Filament tries to upload to S3 directly from the browser instead of doing it server side. Obviously, it doesn't work.

Obviously? It does work for most people? What's your problem?

**Reply by NicolasSacC** [reply]
Didn't mean to offend you, sorry. I thought that it was a mistake that it sent it like that. It just seems to not work on localhost. I will check my settings and see what's wrong.

**Reply by danharrin** [reply]
You didn't offend me, but "obviously" suggests that it wasn't deliberate haha, Livewire built it like that on purpose

Any errors?

**Reply by NicolasSacC** [reply]
It was a CORS issue on AWS.

Maybe it would be a good idea to explain it a bit more in the docs? Since people might expect it to work like the Storage facade which doesn't require these changes in the AWS configuration.

I don't know if I am good enough at writing docs, but maybe I will try adding that :)

**Reply by danharrin** [reply]
Have you fixed the CORS policy now?

**Reply by NicolasSacC** [reply]
It's not ideal, but it's only for dev.

Also, it just doesn't work if I don't set the visibility to "private" manually on all uploads, it's a bit weird but at least we know the fix haha!

```
[
    {
        "AllowedHeaders": [
            "*"
        ],
        "AllowedMethods": [
            "GET",
            "PUT",
            "POST",
            "DELETE"
        ],
        "AllowedOrigins": [
            "*"
        ],
        "ExposeHeaders": []
    }
]
```

**Reply by aurawindsurfing** [reply]
Those policies on AWS are such a pain in the .....
